### PR TITLE
Add getAllTasks query inside rtk query api

### PR DESCRIPTION
**Pull request by faysal-alam-sazim**
### Description of Change
- Added a query inside rtk api
- Query is for get all tasks
### Associated Ticket Link:
- https://trello.com/c/esZLZRWr

### CODE REVIEW

File: shared/redux/store.ts
Comment (ishmam-iqbal-sazim) [inline shared/redux/store.ts:2]:
```suggestion
import { configureStore } from "@reduxjs/toolkit";

import { api } from "./rtk-apis/tasksAPI";
```

Should be a line gap between library and relative imports

File: shared/components/TaskModal/TaskModal.tsx
Comment (ishmam-iqbal-sazim) [inline shared/components/TaskModal/TaskModal.tsx:47]: Why are we only validating due date? Please add a ticket to address form validation.

**Review by ishmam-iqbal-sazim — APPROVED** [review]
Approved pending comments.

**Review by ragib-hossain-sazim — APPROVED** [review]
This PR is not rebased properly, there is a lot of unrelated changes. Please rebase before merging.

**Review by ragib-hossain-sazim — CHANGES_REQUESTED** [review]
Should this be pointed to main?

**Comment by faysal-alam-sazim** [comment]
> Should this be pointed to main?

No this PR should point at the latest branch from this created. I forgot change the pointing branch. But now the latest branch is merged with main. It now should point at main.

**Review by ragib-hossain-sazim — APPROVED** [review]

**Pull request merged**
